File: implementations/javascript/matrix_multiplication/matrix_multiplication_large.js

const readline = require('readline');

const N = 500;
const a = Array.from({ length: N }, () => new Array(N).fill(0));
const b = Array.from({ length: N }, () => new Array(N).fill(0));
const c = Array.from({ length: N }, () => new Array(N).fill(0));

function setup() {
    for (let i = 0; i < N; i++) {
        for (let j = 0; j < N; j++) {
            a[i][j] = i + j;
            b[i][j] = i - j;
        }
    }
}

function matrix_mul() {
    for (let i = 0; i < N; i++) {
        for (let j = 0; j < N; j++) {
            let acc = 0;
            for (let k = 0; k < N; k++) {
                acc += a[i][k] * b[k][j];
            }
            c[i][j] = acc;
        }
    }
    return c[Math.floor(N / 2)][Math.floor(N / 2)];
}

setup();
process.stdout.write("ready\n");

const rl = readline.createInterface({ input: process.stdin });
rl.on('line', () => {
    process.stdout.write(String(matrix_mul()) + "\n");
});

File: implementations/rust/matrix_multiplication/large/matrix_multiplication_large.rs
use std::io::{self, BufRead, Write};

const N: usize = 500;

static mut A: [[i64; N]; N] = [[0; N]; N];
static mut B: [[i64; N]; N] = [[0; N]; N];
static mut C: [[i64; N]; N] = [[0; N]; N];

fn setup() {
    unsafe {
        for i in 0..N {
            for j in 0..N {
                A[i][j] = (i + j) as i64;
                B[i][j] = (i as i64) - (j as i64);
            }
        }
    }
}

fn matrix_mul() -> i64 {
    unsafe {
        for i in 0..N {
            for j in 0..N {
                let mut acc: i64 = 0;
                for k in 0..N {
                    acc += A[i][k] * B[k][j];
                }
                C[i][j] = acc;
            }
        }
        C[N / 2][N / 2]
    }
}

fn main() {
    setup();

    println!("ready");
    io::stdout().flush().unwrap();

    let stdin = io::stdin();
    for _ in stdin.lock().lines() {
        println!("{}", matrix_mul());
        io::stdout().flush().unwrap();
    }
}